tests/server.py

```diff
@@ -190,24 +190,34 @@ def make_request(
     return req, channel
 
 
-def wait_until_result(clock, request, timeout=100):
+def wait_until_result(clock, request):
     """
     Wait until the request is finished.
     """
+    # this function used to give up if the request hadn't completed within 100
+    # iterations. However, this led to problems with log contexts.
+    #
+    # The problem is that because we (a) give up waiting for the result of the
+    # request, and (b) throw away the HomeServer instance (and hence the Notifier,
+    # which would otherwise keep references to requests it is supposed to be
+    # waking up), we can end up with orphaned chains of incomplete Deferreds
+    # which no longer have any references to them and so get garbage-collected.
+    #
+    # It further turns out that one of the many unpleasant effects of the way we
+    # do logcontexts is that if a deferred chain gets garbage-collected before
+    # it completes, we can end up leaking log-contexts (see log_contexts.rst
+    # for more details).
+    #
+    # In short, a timeout here makes things explode down the line.
```

Comment:
Lack of a timeout here causes an infinite loop on jammed web requests in tests...

Reply:
doesn't the test have an overall timeout?

Reply:
The timeout is done by the global reactor, this here does not yield to the global reactor, and therefore can run infinitely and not trip the global reactor timeout.

Reply:
well, that's sad, but given that if we're in that situation the test should have failed, is it really a problem?